erdjs/src/transaction.ts

```diff
@@ -227,10 +325,17 @@ export class Transaction implements ISignable {
             throw new errors.ErrTransactionHashUnknown();
         }
 
-        let response = await provider.getTransaction(this.hash);
+        // For Smart Contract transactions, wait for their full execution & notarization before returning.
```

Comment:
Indeed, quite a workaround - we wait for notarization so that all relevant SCRs are returned by the `getTransaction with results` endpoint. Can we keep as it is for now?

Reply:
I was thinking we could wait for notarization for all transactions not just for smart contract transactions. But this can remain as is though.